presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java

```diff
@@ -1012,6 +1015,81 @@ public static long lastDayOfMonthFromDate(@SqlType(StandardTypes.DATE) long date
         return MILLISECONDS.toDays(millis);
     }
 
+    @Description("Day of the specified day-of-week which is closest and larger than the given date")
+    @ScalarFunction("next_day")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.DATE)
+    public static long nextDayFromDate(@SqlType(StandardTypes.DATE) long date, @SqlType("varchar(x)") Slice dayOfWeek)
+    {
+        long millis = computeNextDay(UTC_CHRONOLOGY, DAYS.toMillis(date), dayOfWeek);
+        return MILLISECONDS.toDays(millis);
+    }
+
+    @Description("Day of the specified day-of-week which is closest and larger than the given timestamp")
+    @ScalarFunction("next_day")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.DATE)
+    public static long nextDayFromTimestamp(ConnectorSession session, @SqlType(StandardTypes.TIMESTAMP) long timestamp, @SqlType("varchar(x)") Slice dayOfWeek)
+    {
+        if (session.isLegacyTimestamp()) {
+            long millis = computeNextDay(getChronology(session.getTimeZoneKey()), timestamp, dayOfWeek);
+            return MILLISECONDS.toDays(millis);
+        }
+        long millis = computeNextDay(UTC_CHRONOLOGY, timestamp, dayOfWeek);
+        return MILLISECONDS.toDays(millis);
+    }
+
+    @Description("Day of the specified day-of-week which is closest and larger than the given timestamp")
+    @ScalarFunction("next_day")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.DATE)
+    public static long nextDayFromTimestamp(@SqlType(StandardTypes.TIMESTAMP_WITH_TIME_ZONE) long timestampWithTimeZone, @SqlType("varchar(x)") Slice dayOfWeek)
```

Comment:
Since the function returns date, can we accept date only as an input?

@electrum

Reply:
It's a good idea.
However last_day_of_month accepts timestamp and date but returns date only. Should we be the same.